### DISCUSSION TITLE
Batching is not suported?

### DISCUSSION THREAD

**Discussion by DavidGOrtega** [snakers4/silero-vad#410] [Q&A]
Im looking at the code and evidently batching can not be supported as the inference seems to be updating 'h', 'c', 'sr' being reused in the next inference. At most I presume that batches should be done at chunks of time level, right?

**Comment by snakers4** [comment]
Technically it is possible. But we do not share this code, becase for some reason it is hard for people to wrap their heads around it.
Also we have a somewhat sophisticated function that takes many "hacks" into consideration when processing a single audio, porting this to a  batch setting would be a chore.

Though ofc course without clever post-processing it is much easier.

> At most I presume that batches should be done at chunks of time level, right?

Right. There are discussions where people experiment with this stuff.
